src/Microsoft.AspNet.Authentication.Facebook/FacebookAuthenticationHandler.cs

```diff
@@ -51,10 +51,10 @@ public FacebookAuthenticationHandler(HttpClient httpClient)
 
         protected override async Task<AuthenticationTicket> CreateTicketAsync(ClaimsIdentity identity, AuthenticationProperties properties, OAuthTokenResponse tokens)
         {
-            var endpoint = Options.UserInformationEndpoint + "?access_token=" + UrlEncoder.UrlEncode(tokens.AccessToken);
+            var endpoint = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, "access_token", UrlEncoder.UrlEncode(tokens.AccessToken));
```

Comment:
When using `QueryHelpers.AddQueryString`, `key` and `value` are automatically url-encoded for you.
Pre-encoding them will result in a terrible double encoding. It's probably not a big deal since access tokens issued by Facebook only use alphanumerical chars, but it's not a guarantee and it may change in the future.

https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNet.WebUtilities/QueryHelpers.cs#L59-L62